crypto/src/address.rs

```diff
@@ -128,8 +144,9 @@ impl std::str::FromStr for Address {
     type Err = anyhow::Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let inner = bech32str::decode(s, bech32str::address::BECH32_PREFIX, bech32str::Bech32m)?;
         pb::Address {
-            inner: bech32str::decode(s, bech32str::address::BECH32_PREFIX, bech32str::Bech32m)?,
+            inner: f4jumble_inv(&inner).ok_or(std::fmt::Error)?,
```

Comment:
I'm not sure this is the right place for the jumbling, because it's only being applied to the displayed bytes, not to the proto-encoded data.

Instead, the jumbling should happen at the boundary between the domain type and the proto type, so that the proto type always records jumbled bytes, and we don't need any special-cased paths.